source/cipher_openssl.c

```diff
@@ -134,17 +135,27 @@ size_t aws_cryptosdk_md_size(enum aws_cryptosdk_md_alg md_alg) {
 }
 
 int aws_cryptosdk_md_update(struct aws_cryptosdk_md_context *md_context, const void *buf, size_t length) {
+    AWS_PRECONDITION(aws_cryptosdk_md_context_is_valid(md_context));
+    AWS_PRECONDITION(buf);
+    AWS_PRECONDITION(AWS_MEM_IS_READABLE(buf, length));
+
     if (1 != EVP_DigestUpdate(md_context->evp_md_ctx, buf, length)) {
+        AWS_POSTCONDITION(aws_cryptosdk_md_context_is_valid(md_context));
         return aws_raise_error(AWS_CRYPTOSDK_ERR_CRYPTO_UNKNOWN);
     }
 
+    AWS_POSTCONDITION(aws_cryptosdk_md_context_is_valid(md_context));
     return AWS_OP_SUCCESS;
 }
 
 int aws_cryptosdk_md_finish(struct aws_cryptosdk_md_context *md_context, void *output_buf, size_t *length) {
+    AWS_PRECONDITION(aws_cryptosdk_md_context_is_valid(md_context));
+    AWS_PRECONDITION(output_buf);
```

Comment:
```AWS_MEM_IS_WRITABLE(output_buf, *length```

Reply:
Fixed, and also changed the harness to match.